src/main/java/io/stargate/sgv2/jsonapi/metrics/MetricsTenantDeactivationConsumer.java

```diff
@@ -0,0 +1,61 @@
+package io.stargate.sgv2.jsonapi.metrics;
+
+import static io.stargate.sgv2.jsonapi.metrics.MetricsConstants.MetricTags.SESSION_TAG;
+import static io.stargate.sgv2.jsonapi.metrics.MetricsConstants.MetricTags.TENANT_TAG;
+
+import com.github.benmanes.caffeine.cache.RemovalCause;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.stargate.sgv2.jsonapi.service.cqldriver.CQLSessionCache;
+import java.util.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * A {@link CQLSessionCache.DeactivatedTenantConsumer} responsible for removing tenant-specific
+ * metrics from the {@link MeterRegistry} when a tenant's session is evicted from the {@link
+ * CQLSessionCache}.
+ */
+public class MetricsTenantDeactivationConsumer
+    implements CQLSessionCache.DeactivatedTenantConsumer {
+  private static final Logger LOGGER =
+      LoggerFactory.getLogger(MetricsTenantDeactivationConsumer.class);
+  private final MeterRegistry meterRegistry;
+
+  public MetricsTenantDeactivationConsumer(MeterRegistry meterRegistry) {
+    Objects.requireNonNull(meterRegistry, "MeterRegistry cannot be null.");
+    this.meterRegistry = meterRegistry;
```

Comment:
suggestion: can combine 2 lines into:

   this.meterRegistry = Objects.requireNonNull(meterRegistry, "MeterRegistry cannot be null");